toolchain/sem_ir/value_stores.h

```diff
@@ -43,6 +43,48 @@ class NodeStore {
   ValueStore<NodeId, Node> values_;
 };
 
+// Provides a ValueStore-like interface for names. This is currently a wrapper
+// around a string store that has no state of its own.
+class NameStore {
+ public:
+  NameStore(ValueStore<StringId>* strings) : strings_(strings) {}
+
+  // Adds a string name, returning an ID to reference it.
+  auto Add(StringId string_id) -> NameId { return NameId(string_id.index); }
+
+  // Returns the requested name as a string ID, if it is a string name. This
+  // returns std::nullopt for special names.
+  auto GetAsStringId(NameId name_id) const -> std::optional<StringId> {
+    if (name_id.index >= 0) {
+      return StringId(name_id.index);
+    }
+    return std::nullopt;
+  }
+
+  // Returns the requested name as a string, if it is a string name. This
+  // returns std::nullopt for special names.
+  auto GetAsString(NameId name_id) const -> std::optional<llvm::StringRef> {
```

Comment:
"GetAsString" is something I expect to return `std::string`. Maybe `GetIfInStrings`, `GetIfNotSpecial`, `GetFromStrings`, `Get`, or similar?

Reply:
Done, `GetIfIdentifier`, plus extra comments to explain identifier names versus special names.